### PR TITLE
Use AssertIndexRange some more

### PR DESCRIPTION
**Pull request by masterleinad**
Mostly generated by
```
find include/ source/ \( -name "*.cc" -o -name "*.h" \) -exec perl -0777 -i.original -pe 's/Assert\(([^<]*)[\s]*<([^,]*),[^E]*ExcIndexRange[^(]*\([\s]*\1,[\s]*0,[\s]*\2[\s]*\)[\s]*\);/AssertIndexRange(\1,\2);/igs' {} \;
```

### CODE REVIEW

File: tests/base/functions_04.cc [hunk 1]
```diff
@@ -217,6 +217,8 @@ main()
     {
       deallog << " Functions::StokesCosine<3>" << std::endl;
       Functions::StokesCosine<3> f(1.);
+      Functions::StokesCosine<3> g = f;
+      g                            = f;
```
Comment (kronbichler) [inline tests/base/functions_04.cc]: I guess this change is spurious?

Comment (masterleinad) [reply tests/base/functions_04.cc]: Yes, sorry.

**Comment by kronbichler** [comment]
@masterleinad can you please rebase?

**Comment by masterleinad** [comment]
> @masterleinad can you please rebase?

Done.

**Comment by bangerth** [comment]
It currently fails like this:
```
/jenkins/workspace/dealii_PR-9268/include/deal.II/base/geometry_info.h: In static member function ‘static unsigned int dealii::GeometryInfo<dim>::face_to_cell_lines(unsigned int, unsigned int, bool, bool, bool) [with int dim = 1]’:

/jenkins/workspace/dealii_PR-9268/include/deal.II/base/exceptions.h:1645:32: error: comparison of unsigned expression < 0 is always false [-Werror=type-limits]

                   type>(index) <                                               \

                                ^

/jenkins/workspace/dealii_PR-9268/include/deal.II/base/exceptions.h:1413:32: note: in definition of macro ‘Assert’

         if (__builtin_expect(!(cond), false))                            \

                                ^

/jenkins/workspace/dealii_PR-9268/include/deal.II/base/geometry_info.h:4243:3: note: in expansion of macro ‘AssertIndexRange’

   AssertIndexRange(line, lines_per_face);

   ^
```

**Comment by luca-heltai** [comment]
Please squash, then merge yourself.

**Pull request merged**